### PR TITLE
Pagination for direct model lookup calls

### PR DESCRIPTION
**Pull request by rranjan3**
Enable lookup next calls for registry/worker/receipt lookup.

### CODE REVIEW

File: client_sdk/avalon_client_sdk/ethereum/ethereum_worker_registry_list_client.py [hunk 1]
```diff
@@ -33,11 +33,16 @@ class EthereumWorkerRegistryListClientImpl(WorkerRegistryListClient):
     This class is to read worker registries enttries from Ethereum.
```
Comment (danintel) [inline client_sdk/avalon_client_sdk/ethereum/ethereum_worker_registry_list_client.py]: s/enttries/entries/

File: client_sdk/avalon_client_sdk/ethereum/ethereum_worker_registry_list_client.py [hunk 2]
```diff
@@ -144,6 +149,19 @@ def registry_set_status(self, org_id, status):
...
     def registry_lookup(self, app_type_id=None):
+        return self.__registry_lookup(app_type_id, None, False)
+
+    def registry_lookup_next(self, app_type_id, lookup_tag):
+        return self.__registry_lookup(app_type_id, lookup_tag, True)
+
+    def __registry_lookup(self, app_type_id=None, lookup_tag=None, lookup_next=False):
+        """
+        This function handles both the lookup & lookup_next calls. It looks
+        for the flag, lookup_next which if enabled, it would discard all
```
Comment (danintel) [inline client_sdk/avalon_client_sdk/ethereum/ethereum_worker_registry_list_client.py]: s/lookup_next/lookup_next,/

File: client_sdk/avalon_client_sdk/ethereum/ethereum_worker_registry_list_client.py [hunk 3]
```diff
@@ -144,6 +149,19 @@ def registry_set_status(self, org_id, status):
...
+        entries in result set till the look_up tag. It paginates results
```
Comment (danintel) [inline client_sdk/avalon_client_sdk/ethereum/ethereum_worker_registry_list_client.py]: s/in result set till/in the result set until/

File: client_sdk/avalon_client_sdk/ethereum/ethereum_worker_registry_list_client.py [hunk 4]
```diff
@@ -144,6 +149,19 @@ def registry_set_status(self, org_id, status):
...
+        post this ans returns as next page.
```
Comment (danintel) [inline client_sdk/avalon_client_sdk/ethereum/ethereum_worker_registry_list_client.py]: s/post this ans returns as next page./after this and returns it as the next page.

File: client_sdk/avalon_client_sdk/tcf_connector.toml [hunk 5]
```diff
@@ -21,6 +21,7 @@ organization = "Intel"
 [tcf]
 #Uri for the worker registry
 json_rpc_uri = "http://localhost:1947"
+lookup_page_size = 20
```
Comment (danintel) [inline client_sdk/avalon_client_sdk/tcf_connector.toml]: Add a short comment explaining what this is used for.  Something like:

`# Paginate results into lookup_page_size items for registry_lookup() and registry_lookup_next() calls`

(change the above if it's not correct)

File: client_sdk/avalon_client_sdk/ethereum/ethereum_worker_registry_list_client.py [hunk 6]
```diff
@@ -160,7 +178,26 @@ def registry_lookup(self, app_type_id=None):
                 lookupResult = \
                     self.__contract_instance.functions.registryLookUp(b"") \
                     .call()
-            return lookupResult
+            wCount, lookUpString, wList = lookupResult
```
Comment (manojgop) [inline client_sdk/avalon_client_sdk/ethereum/ethereum_worker_registry_list_client.py]: Please use PEP8 naming convention for variables. eg: lookupString -> lookup_string, lookupResult -> lookup_result

File: client_sdk/avalon_client_sdk/ethereum/ethereum_worker_registry_list_client.py [hunk 7]
```diff
@@ -160,7 +178,26 @@ def registry_lookup(self, app_type_id=None):
...
+            if lookup_next:
+                residue_list = []
+                for reg in wlist:
+                    if reg.orgId != lookup_tag:
```
Comment (manojgop) [inline client_sdk/avalon_client_sdk/ethereum/ethereum_worker_registry_list_client.py]: So is lookup_tag corresponds to org ID. It's not very obvious to follow the code. Comments will help

File: client_sdk/avalon_client_sdk/ethereum/ethereum_worker_registry_list_client.py [hunk 8]
```diff
@@ -160,7 +178,26 @@ def registry_lookup(self, app_type_id=None):
...
+                        wCount -= 1
+                        continue
+                    residue_list.append(reg)
+                wlist = residue_list
+
+            if wCount > self.lookup_page_size:
```
Comment (manojgop) [inline client_sdk/avalon_client_sdk/ethereum/ethereum_worker_registry_list_client.py]: Please add comment for this code block on what is this trying to do

Comment (rranjan3) [reply client_sdk/avalon_client_sdk/ethereum/ethereum_worker_registry_list_client.py]: Addressed all your comments.

File: client_sdk/avalon_client_sdk/tcf_connector.toml (hunk 5 above)
Comment (rranjan3) [reply client_sdk/avalon_client_sdk/tcf_connector.toml]: Thanks Dan. Addressed all.

File: config/tcs_config.toml [hunk 9]
```diff
@@ -32,6 +32,7 @@ sleep_interval = "10"
 [Listener]
 listener_port = 1947
 max_work_order_count = 10
+lookup_page_size = 20
```
Comment (Ram-srini) [inline config/tcs_config.toml]: Do we need this constant here also? It is already present in tcf_connector.toml.

Comment (rranjan3) [reply config/tcs_config.toml]: Yes @Ram-srini, we would need it here as well. The listener handlers make use of this config file.

File: client_sdk/avalon_client_sdk/ethereum/ethereum_worker_registry_list_client.py [hunk 10]
```diff
@@ -30,9 +30,11 @@
 
 class EthereumWorkerRegistryListClientImpl(WorkerRegistryListClient):
     """
-    This class is to read worker registries enttries from Ethereum.
+    This class is to read worker registries entries from Ethereum.
     Implements WorkerRegistryListClient interface
     """
+    lookup_page_size = 20 # Assigning a default size if config absent
```
Comment (manojgop) [inline client_sdk/avalon_client_sdk/ethereum/ethereum_worker_registry_list_client.py]: make this a class member variable and initialize it in _initialize() function

File: client_sdk/avalon_client_sdk/ethereum/ethereum_worker_registry_list_client.py [hunk 11]
```diff
@@ -144,11 +146,24 @@ def registry_set_status(self, org_id, status):
...
+    def registry_lookup_next(self, app_type_id, lookup_tag):
+        return self.__registry_lookup(app_type_id, lookup_tag, True)
+
+    def __registry_lookup(self, app_type_id=None, lookup_tag=None, lookup_next=False):
+        """
+        This function handles both the lookup & lookup_next calls. It looks
+        for the flag, lookup_next, which if enabled, it would discard all
+        entries in the result set until the look_up tag. It paginates results
+        post this and returns it as the next page.
+        """
```
Comment (manojgop) [inline client_sdk/avalon_client_sdk/ethereum/ethereum_worker_registry_list_client.py]: Please add a comment on how lookup_tag is used

**Comment by rranjan3** [comment]
Another PR to be raised with cursor based pagination at lmdb.

**Pull request closed without merging**